scene/gui/tree.cpp

```diff
@@ -315,6 +315,9 @@ void TreeItem::set_text(int p_column, String p_text) {
 			cells.write[p_column].max = MAX(cells[p_column].max, value);
 		}
 		cells.write[p_column].step = 0;
+	} else {
+		// Don't auto translate if it's in string mode and editable, as the text can be changed to anything by the player.
+		cells.write[p_column].xl_text = tree && (!cells[p_column].editable || cells[p_column].mode != TreeItem::CELL_MODE_STRING) ? tree->atr(p_text) : p_text;
```

Comment:
Chaining an assignment with a second statement like this looks pretty weird and hard to read.

Reply:
Oh well I misread (on phone), it's a convoluted ternary assignment. I would add an extra pair of parentheses around the full condition, or make this a plain `if () {} else {}`.